File: Moobeez/Classes/Moobeez/Views/SearchResultCell.m
//
//  SearchResultCell.m
//  Moobeez
//
//  Created by Radu Banea on 01/11/13.
//  Copyright (c) 2013 Goggzy. All rights reserved.
//

#import "SearchResultCell.h"
#import "Moobeez.h"

@interface SearchResultCell ()

@property (weak, nonatomic) IBOutlet ImageView *posterImageView;

@property (weak, nonatomic) IBOutlet UILabel *titleLabel;

@property (weak, nonatomic) IBOutlet UILabel *detailsLabel;

@end

@implementation SearchResultCell

- (void)awakeFromNib {
    [super awakeFromNib];
    
    self.posterImageView.loadSyncronized = YES;
}

- (void)setTmdbMovie:(TmdbMovie *)tmdbMovie {
    _tmdbMovie = tmdbMovie;
    
    [self.posterImageView loadPosterWithPath:tmdbMovie.posterPath completion:^(BOOL didLoadImage) {}];
    self.titleLabel.text = tmdbMovie.name;
    self.detailsLabel.text = [[NSDateFormatter dateFormatterWithFormat:@"dd MMM yyyy"] stringFromDate:tmdbMovie.releaseDate];
    
    self.detailsLabel.hidden = (self.detailsLabel.text.length == 0);
}

- (void)setTmdbPerson:(TmdbPerson *)tmdbPerson {
    _tmdbPerson = tmdbPerson;
    
    [self.posterImageView loadProfileWithPath:tmdbPerson.profilePath completion:^(BOOL didLoadImage) {}];
    self.titleLabel.text = tmdbPerson.name;
    self.detailsLabel.text = tmdbPerson.overview;
    
    self.detailsLabel.hidden = (self.detailsLabel.text.length == 0);
}

- (void)setTmdbTv:(TmdbTV *)tmdbTv {
    _tmdbTv = tmdbTv;
    
    [self.posterImageView loadPosterWithPath:tmdbTv.posterPath completion:^(BOOL didLoadImage) {}];
    self.titleLabel.text = tmdbTv.name;
    self.detailsLabel.text = [[NSDateFormatter dateFormatterWithFormat:@"dd MMM yyyy"] stringFromDate:tmdbTv.releaseDate];
    
    self.detailsLabel.hidden = (self.detailsLabel.text.length == 0);
}


@end

File: Moobeez/Classes/UI/General/Cells/Search Result Cell/SearchResultCell.swift
//
//  SearchResultCell.swift
//  Moobeez
//
//  Created by Radu Banea on 09/10/2017.
//  Copyright © 2017 Gogo's Lab. All rights reserved.
//

import UIKit

class SearchResultCell: UITableViewCell {

    @IBOutlet var posterImageView:UIImageView!
    
    @IBOutlet var titleLabel:UILabel!
    
    @IBOutlet var detailsLabel:UILabel!
    
    override func awakeFromNib() {
        super.awakeFromNib()
        // Initialization code
    }
    
    var movie:Tmdb.Movie?
    {
        didSet {
            posterImageView.loadTmdbPosterWithPath(path: movie?.posterPath, placeholder:#imageLiteral(resourceName: "default_image"))
            titleLabel.text = movie?.name
            
            if movie?.releaseDate != nil {
                let dateFormatter = DateFormatter()
                dateFormatter.dateFormat = "dd MMM yyyy"
                detailsLabel.text = dateFormatter.string(from: (movie?.releaseDate)!)
            }
            else {
                detailsLabel.text = ""
            }
            
            detailsLabel.isHidden = (detailsLabel.text?.count == 0)
        }
    }
    
    var person:Tmdb.Person?
    {
        didSet {
            posterImageView.loadTmdbProfileWithPath(path: person?.profilePath)
            titleLabel.text = person?.name
            
            detailsLabel.text = person?.overview ?? ""
            
            detailsLabel.isHidden = (detailsLabel.text?.count == 0)
        }
    }
    
    var tvShow:Tmdb.TvShow?
    {
        didSet {
            posterImageView.loadTmdbPosterWithPath(path: tvShow?.posterPath, placeholder:#imageLiteral(resourceName: "default_image"))
            titleLabel.text = tvShow?.name
            
            if tvShow?.releaseDate != nil {
                let dateFormatter = DateFormatter()
                dateFormatter.dateFormat = "dd MMM yyyy"
                detailsLabel.text = dateFormatter.string(from: (tvShow?.releaseDate)!)
            }
            else {
                detailsLabel.text = ""
            }
        }
    }

}
